recipes/openimageio/all/patches/2.2.7.0-cmake-targets.patch

```diff
@@ -145,20 +523,151 @@ index 083b077c..a7ec5cda 100644
+-                     LINK_LIBRARIES ${PTEX_LIBRARIES} ${ZLIB_LIBRARIES}
++                     # INCLUDE_DIRS ${ZLIB_INCLUDE_DIRS} ${PTEX_INCLUDE_DIR}
++                     LINK_LIBRARIES PTex::PTex ZLIB::ZLIB
+                      DEFINITIONS "-DUSE_PTEX")
+ endif ()
 diff --git a/src/raw.imageio/CMakeLists.txt b/src/raw.imageio/CMakeLists.txt
-index 81a0ff54..7be59ed7 100644
+index 81a0ff54..ae229e31 100644
 --- a/src/raw.imageio/CMakeLists.txt
 +++ b/src/raw.imageio/CMakeLists.txt
 @@ -2,10 +2,10 @@
  # SPDX-License-Identifier: BSD-3-Clause
  # https://github.com/OpenImageIO/oiio/blob/master/LICENSE.md
  
 -if (LIBRAW_FOUND)
-+if (TARGET libraw::libraw)
++if (libraw_FOUND)
      add_oiio_plugin (rawinput.cpp
-                      INCLUDE_DIRS ${LibRaw_INCLUDE_DIR}
+-                     INCLUDE_DIRS ${LibRaw_INCLUDE_DIR}
 -                     LINK_LIBRARIES ${LibRaw_r_LIBRARIES}
++                     # INCLUDE_DIRS ${LibRaw_INCLUDE_DIR}
 +                     LINK_LIBRARIES libraw::libraw
                       DEFINITIONS "-DUSE_LIBRAW=1")
  else ()
      message (WARNING "Raw plugin will not be built")
+diff --git a/src/tiff.imageio/CMakeLists.txt b/src/tiff.imageio/CMakeLists.txt
+index f6d648c8..385607c9 100644
+--- a/src/tiff.imageio/CMakeLists.txt
++++ b/src/tiff.imageio/CMakeLists.txt
+@@ -3,6 +3,6 @@
+ # https://github.com/OpenImageIO/oiio/blob/master/LICENSE.md
+ 
+ add_oiio_plugin (tiffinput.cpp tiffoutput.cpp
+-                 INCLUDE_DIRS ${TIFF_INCLUDE_DIR} ${ZLIB_INCLUDE_DIRS}
+-                 LINK_LIBRARIES ${TIFF_LIBRARIES} ${JPEG_LIBRARIES}
+-                                ${ZLIB_LIBRARIES})
++                 # INCLUDE_DIRS ${TIFF_INCLUDE_DIR} ${ZLIB_INCLUDE_DIRS}
++                 LINK_LIBRARIES TIFF::TIFF JPEG::JPEG
++                                ZLIB::ZLIB)
+diff --git a/src/webp.imageio/CMakeLists.txt b/src/webp.imageio/CMakeLists.txt
+index 9e01bd7f..78fe1a58 100644
+--- a/src/webp.imageio/CMakeLists.txt
++++ b/src/webp.imageio/CMakeLists.txt
+@@ -2,9 +2,9 @@
+ # SPDX-License-Identifier: BSD-3-Clause
+ # https://github.com/OpenImageIO/oiio/blob/master/LICENSE.md
+ 
+-if (Webp_FOUND)
++if (WebP_FOUND)
+     add_oiio_plugin (webpinput.cpp webpoutput.cpp
+-                     LINK_LIBRARIES Webp::Webp
++                     LINK_LIBRARIES WebP::WebP
+                      DEFINITIONS "-DUSE_WEBP=1")
+ else ()
+     message (STATUS "WebP plugin will not be built")
+diff --git a/src/zfile.imageio/CMakeLists.txt b/src/zfile.imageio/CMakeLists.txt
+index 36a7ad99..c8fd99d3 100644
+--- a/src/zfile.imageio/CMakeLists.txt
++++ b/src/zfile.imageio/CMakeLists.txt
+@@ -3,5 +3,5 @@
+ # https://github.com/OpenImageIO/oiio/blob/master/LICENSE.md
+ 
+ add_oiio_plugin (zfile.cpp
+-                 INCLUDE_DIRS ${ZLIB_INCLUDE_DIRS}
+-                 LINK_LIBRARIES ${ZLIB_LIBRARIES})
++                 # INCLUDE_DIRS ${ZLIB_INCLUDE_DIRS}
++                 LINK_LIBRARIES ZLIB::ZLIB)
```

Comment:
Not much to do here. All patches are big, but they make all dependencies work correctly with Conan.  :+1:

Reply:
Upstream could make our life somewhat easier if they used more target-based linking with standard names...